### PR TITLE
chore: add api test structure

### PR DESCRIPTION
**Pull request by timstackblock**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/2364214686).

This PR creates a separate test build path and test directories to run browser automated test against the API network URL https://stacks-node-api-inactive.stacks.co/extended/v1/tx

There are 8 test added to the Test-Api directory that test functionality around 

- Balance
- Adding a network
- Sending STX
- Funding the account

These test will be improved and updated as my main focus of this PR was making sure the framework was working as expected.

To run the test do the following:
yarn install 
yarn build:test-api 
yarn test:integration-api

cc/ @Eshwari007 @kyranjamie @fbwoolf @beguene @He1DAr @CharlieC3 @saralab @rafaelcr @zone117x 

Next steps will be to put the URL https://stacks-node-api-inactive.stacks.co/extended/v1/tx into an environment variable and have the scripts read the environment variable. This will clear the path for when the API URL is created by Vercel and passed into an environment variable. Our goal is to build a script that can pick that variable up and use it for the test.

https://github.com/hirosystems/stacks-blockchain-api/issues/1002

### CODE REVIEW

**Comment by timstackblock** [comment]
I have some linting issues I need to address before I can merge this I will work on resolving these.

**Comment by andresgalante** [comment]
@timstackblock It might be a good idea for you to install prettier in your IDE, to catch linting issues before sending a PR. Most of the time it even auto-fixes them on save ✨

File: tests/test-api/integration/buy-token/buy-tokens.spec.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  }, BEFORE_EACH_TIMEOUT);
+
+  afterEach(async () => {
+    try {
+      await browser.context.close();
+    } catch (error) {}
+  });
+
+  describe('Transak test', () => {
+    it('click button should redirect to transak URL', async () => {
```
Comment (kyranjamie) [inline tests/test-api/integration/buy-token/buy-tokens.spec.ts:35]: try and keep tests readable:

```suggestion
    it('should redirect to transak URL on button click', async () => {
```

File: tests/test-api/integration/send-tokens/send-tokens.spec.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { RouteUrls } from '@shared/route-urls';
+import { SECRET_KEY_2 } from '@tests/mocks';
+
+import { SendPage } from '../../../page-objects/send-form.page';
+import { WalletPage } from '../../../page-objects/wallet.page';
+import {
+  BrowserDriver,
+  setupBrowser,
+  addAPINetwork
+} from '../../../integration/utils';
```
Comment (kyranjamie) [inline tests/test-api/integration/send-tokens/send-tokens.spec.ts]: Use `@tests` alias

**Review by kyranjamie — APPROVED** [review]
Good work, @timstackblock, these tests all look good to me.

Just an idea, but next time to help get a feel for what the tests are doing, it'd be really helpful to include a recording of the tests running in the PR.

**Comment by timstackblock** [comment]
I have updated the remaining test only the settings test still need modifications.
To point these test against 

yarn install
yarn build:test-api 
APIEnvVariable=<some_url> yarn test:integration-api

**Pull request closed without merging**